ksqldb-engine/src/main/java/io/confluent/ksql/engine/generic/GenericRecordFactory.java

```diff
@@ -78,7 +78,13 @@ public KsqlGenericRecord build(
     final LogicalSchema schemaWithRowTime = withRowTime(schema);
     for (ColumnName col : columns) {
       if (!schemaWithRowTime.findColumn(col).isPresent()) {
-        throw new KsqlException("Column name " + col + " does not exist.");
+        final String message;
+        if (SystemColumns.disallowedForInsertValues(col)) {
+          message = "Inserting into column " + col + " is not allowed.";
+        } else {
+          message = "Column name " + col + " does not exist.";
+        }
+        throw new KsqlException(message);
```

Comment:
The changes in this file should be accompanied by unit test changes. See GenericRecordFactoryTest.java and InsertValuesExecutorTest.java for inspiration.